apps/slackbot/task_agent.py

```diff
@@ -150,6 +152,29 @@ def run(self, task: str) -> str:
             action = self.output_parser.parse(assistant_reply)
             print("action:", action)
             tools = {t.name: t for t in self.tools}
+            if action == previous_action:
+                if action.name == "Search" or action.name == "Context Search":
+                    print("Action name: ", action.name, "\nStart reformulating the query")
+                    instruction = (
```

Comment:
If I understand correctly, this prompt only asks for a _different input_ for the _same_ tool. Is this intended?